updatecli/weekly.d/jenkinscontroller-tools-jdk21.yaml

```diff
@@ -12,25 +12,19 @@ scms:
       token: "{{ requiredEnv .github.token }}"
       username: "{{ .github.username }}"
       branch: "{{ .github.branch }}"
-  temurin21-binaries:
-    kind: "github"
-    spec:
-      user: "{{ .github.user }}"
-      email: "{{ .github.email }}"
-      owner: "adoptium"
-      repository: "temurin21-binaries"
-      token: '{{ requiredEnv .github.token }}'
-      branch: "main"
 
 sources:
-  getLatestJDK21EAVersion:
+  getLatestJDK21Version:
+    kind: githubrelease
     name: Get the latest Adoptium JDK21 version
-    kind: gittag
-    scmid: temurin21-binaries
     spec:
+      owner: "adoptium"
+      repository: "temurin21-binaries"
+      token: "{{ requiredEnv .github.token }}"
+      username: "{{ .github.username }}"
       versionfilter:
         kind: regex
-        pattern: ".*-ea-.*"
+        pattern: "^jdk-21.(\\d*).(\\d*).(\\d*)+(\\d*)$"
```

Comment:
NB: these dots in the regexp match any character, not the `.` character.

Not a problem, just a remark.

Reply:
we need to match `jdk-21+35` so if we escape the dots, I think we will not catch it, don't you think ?. 
I didn't question the pattern as it is the one used for all our version of jdk but maybe we should.

Reply:
Proposition:
```suggestion
        pattern: "^jdk-21(\\.\\d+)*\\+(\\d+)$"
```
jdk-21 followed by zero to N group of a dot and one to N numbers, followed by a plus and one to N numbers

Reply:
<img width="664" alt="image" src="https://github.com/jenkins-infra/jenkins-infra/assets/91831478/0c042f7c-0a17-4a54-ab2e-0dbc1466884e">

Reply:
I can propose this change in a distinct PR if you prefer.